test/Microsoft.AspNetCore.Mvc.FunctionalTests/HtmlGenerationTest.cs

```diff
@@ -523,6 +523,30 @@ public class HtmlGenerationTest :
             Assert.Equal(expected, response, ignoreLineEndingDifferences: true);
         }
 
+        // We want to make sure that for 'wierd' model expressions involving:
+        // - fields
+        // - statics
+        // - private
+        //
+        // that we don't throw, and can evaluate the expression to get the model value. We don't
+        // expect that the model metadata returned as accurate, so we're not checking that.
```

Comment:
>So, nothing is being lost or not checked.

Exactly, the fact that if **there were metadata attributes on those members, those metadata attributes would be ignored** is not being tested. I'll improve the description so that it's clear the test intentionally omits a detail that we decided to omit.